### ISSUE TITLE
Autoware parks inappropriately on parking lot

### ISSUE THREAD

**Issue by VRichardJP** [autowarefoundation/autoware.universe#2991]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

In this example, I have set the goal point perpendicular to the initial position (in red), so a maneuver is required. Autoware starts its maneuver but kind of stops halfway, as it considers the goal has been reached as soon as the XY matches:
![image](https://user-images.githubusercontent.com/18645627/222606674-ed336b70-482c-4789-b14a-3bbe576622aa.png)

I guess this is because the goal heading error tolerance is too high. For lane driving, it is not a problem as we can't maneuver, but when the vehicle is on a parking lot, Autoware should park precisely.

I guess this should be fixed for cargo ODD.

### Expected behavior

Autoware parks precisely on parking lot

### Actual behavior

heading error is too big on goal.

### Steps to reproduce

Run simulator with a parking lot and ask autoware to park somewhere.

**Comment by yukkysaito** [comment]
Thanks for the report.
The parking planner has not been changed for some time since it was created as a prototype.
The free space planner doesn't generate stable trajectory, so when it stops at a certain place, we create the lanelet.

It may be better to review the design from the beginning if necessary this time with cargo ODD or bus ODD.

**Comment by NorahXiong** [comment]
@VRichardJP Which planning algorithm did you choose? A-star or RRT?

**Comment by VRichardJP** [comment]
@NorahXiong A* in this example

**Comment by NorahXiong** [comment]
I did a similar test with the default settings and I got a pretty good result. 
@VRichardJP Have you repeated the test? Did it stop halfway everytime? 

![image](https://user-images.githubusercontent.com/103234047/225558210-01b62e74-9b5a-46ed-841e-edca8cd59744.png)

**Comment by NorahXiong** [comment]
I did a similar test with the default settings and I got a pretty good result. 
@VRichardJP Have you repeated the test? Did it stop halfway everytime? 

![image](https://user-images.githubusercontent.com/103234047/225558210-01b62e74-9b5a-46ed-841e-edca8cd59744.png)

**Comment by VRichardJP** [comment]
In my case, I added a few obstacles to make the parking maneuver less trivial. You can see the 2 buses left and right of the parking spot in my screenshot. I only tested a couple times with this setup.

**Comment by VRichardJP** [comment]
I am using the same map you are showing.

Starting position:
![image](https://user-images.githubusercontent.com/18645627/226836790-4d058b7a-1f91-4e2d-b197-2ecedef8bbf0.png)
After struggling with the maneuver, going back and forth many times just a few meters, autoware drove very quickly close to the target goal and got stuck:
![image](https://user-images.githubusercontent.com/18645627/226837088-a98a970f-1d6f-40b0-9181-3f345a47f5a4.png)

Another again. Start position:
![image](https://user-images.githubusercontent.com/18645627/226837520-e6ac53f3-2371-4674-b6f6-489e024537d8.png)
After goal is reached:
![image](https://user-images.githubusercontent.com/18645627/226837636-a117f24e-e65c-4580-9a9c-167b43f3b227.png)

As @yukkysaito said, it seems that Autoware considered the goal has been reached if it is stopped around the goal, even if the maneuver is not finished. This occurs typically when Autoware moves back and forth around the goal to adjust its orientation

**Comment by NorahXiong** [comment]
Hi, @VRichardJP 
I updated my environment and reproduced your issue. I reduced the "vehicle_shape_margin_m" value to 0.2(m) and the parking procedure succeeded.
I checked the "vehicle_shape_margin_m" parameter changelog and I found no one modifying the default value recently. The problem may come from the perception module. Maybe algorithms and logic changed in the perception module and it results in unstable/incorrect distance measurement.
Anyway, hope my information could help you!
